src/lit-html.ts

```diff
@@ -220,10 +221,18 @@ export class Template {
             parent.insertBefore(document.createTextNode(strings[i]), node);
             this.parts.push(new TemplatePart('node', index++));
           }
-        } else if (!node.nodeValue!.trim()) {
-          nodesToRemove.push(node);
-          currentNode = previousNode;
-          index--;
+        } else {
+          // Strip whitespace-only nodes, only between elements, or at the
+          // beginning or end of elements.
+          const previousSibling = node.previousSibling;
+          const nextSibling = node.nextSibling;
+          if ((previousSibling === null || previousSibling.nodeType === 1)
```

Comment:
Please indicate what node type this is as you have done elsewhere in the source.

Reply:
or use an enum.

Reply:
added the comment